### PR TITLE
fix(path-alias): resolve relative tsconfig extends against declaring file's directory

### LINKED ISSUE #177: Relative paths in tsconfig `extends` not supported
## Summary

Relative paths in the tsconfig `extends` field don't appear to work. If I adjust the path to be relative to the cwd, things work as expected.

Happy to open a PR if that'd be helpful. Thanks!

## Reproduction steps

Adjust the `extends` field in a tsconfig file to use a relative path. Run Skott, and observe exceptions when reading tsconfig.

  **Expected result:** 

  **Actual result:** 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `skott` installed version? | 0.35.3 |
| Operating system? | Mac |
| Would you consider contributing a PR? | Yes |
| Node.js version (`node -v`)? | v20.11.1 |

### LINKED ISSUE #210: Infinite loop when tsconfig extends a node_modules package that uses relative extends internally (e.g. @vue/tsconfig)
## Summary

Running skott on any standard Vue 3 project hangs indefinitely. The process never terminates — it just keeps printing the same tsconfig resolution lines in a loop.

## Reproduction

Any Vue 3 project scaffolded with `create-vue` reproduces this. The root `tsconfig.json` looks like this (standard output):

```json
{
  "extends": "@vue/tsconfig/tsconfig.dom.json",
  "include": ["src/**/*", "src/**/*.vue"]
}
```

`@vue/tsconfig/tsconfig.dom.json` (inside node_modules) contains:

```json
{ "extends": "./tsconfig.json" }
```

This `./tsconfig.json` is meant to refer to the **sibling** `tsconfig.json` inside the `@vue/tsconfig` package directory. TypeScript resolves it correctly. Skott resolves it against CWD instead, landing on the project-root `tsconfig.json` — which in turn extends `@vue/tsconfig/tsconfig.dom.json` again → infinite loop.

## Observed output (runs forever, never exits)

```
Reading from tsconfig: @vue/tsconfig/tsconfig.dom.json
Found tsconfig extension: ./tsconfig.json
Reading from tsconfig: ./tsconfig.json
✓ Successfully found tsconfig: /project/tsconfig.json
Found tsconfig extension: @vue/tsconfig/tsconfig.dom.json
Reading from tsconfig: @vue/tsconfig/tsconfig.dom.json
... (repeats indefinitely)
```

## Root cause

In `buildPathAliases` (`src/modules/walkers/ecmascript/typescript/path-alias.ts`), when following a `tsconfig.extends` chain the raw extends string is passed directly to the recursive call. Inside `readTSConfig` it is joined against `fileReader.getCurrentWorkingDir()` — always the project root — instead of against the **directory of the tsconfig file that declares the extends**.

When a third-party package is resolved to an absolute path by `require.resolve`, any relative `extends` inside it (e.g. `./tsconfig.json`) should be resolved relative to *that package's directory*, not CWD.

## Related issues

- #177 — *"Relative paths in tsconfig `extends` not supported"* — same underlying path-resolution bug, reported October 2024, still open.
- #142 — *"Can support for vue projects be added?"* — the `@vue/tsconfig` pattern is one of the main reasons skott does not work out of the box on Vue 3 projects.

## Fix

A pull request with a fix and a regression test is available: #211.

The fix has two parts:
1. `readTSConfig` now returns the resolved path (relative or absolute) alongside the parsed config.
2. `buildPathAliases` resolves relative `extends` paths against `path.dirname(resolvedPath)` instead of always using CWD. A visited-configs `Set` is added as a safety net to break any remaining circular chain gracefully.

### PR DESCRIPTION
**Pull request by brugmanjoost**
## Problem

When a `tsconfig.json` extends a package from `node_modules` that itself uses a **relative `extends`** internally, skott enters an **infinite loop** and never terminates.

The most common real-world trigger is [`@vue/tsconfig`](https://github.com/vuejs/tsconfig) — the officially recommended TypeScript base for Vue 3 projects. Its `tsconfig.dom.json` contains `"extends": "./tsconfig.json"`, which is meant to extend its **sibling** inside the package directory. TypeScript resolves this correctly. Skott does not: it resolves the relative path against CWD, landing on the project-root config, which extends `@vue/tsconfig/tsconfig.dom.json` again → infinite loop.

```
Reading from tsconfig: @vue/tsconfig/tsconfig.dom.json
Found tsconfig extension: ./tsconfig.json
Reading from tsconfig: ./tsconfig.json          ← resolved against CWD (wrong)
✓ Successfully found tsconfig: /project/tsconfig.json
Found tsconfig extension: @vue/tsconfig/tsconfig.dom.json
... (repeats forever)
```

Closes #210. Also fixes #177. Related: #142.

## Root cause

In `buildPathAliases`, the raw `extends` string is passed directly to the recursive call. In `readTSConfig` it is then joined against `fileReader.getCurrentWorkingDir()` — always the project root — instead of against the **directory of the tsconfig file that declared the extends**.

## Fix

Two changes in `src/modules/walkers/ecmascript/typescript/path-alias.ts`:

1. **`readTSConfig` now returns the resolved path** alongside the parsed config — the CWD-relative path from the direct-read attempt, and the absolute path from `require.resolve` for third-party packages.

2. **`buildPathAliases` resolves relative extends against the declaring file's directory.** When `resolvedPath` is relative, `path.join(path.dirname(resolvedPath), extends)` is used (preserves the relative form, so in-memory file readers used in tests continue to work). When `resolvedPath` is absolute (real `node_modules` path), `path.resolve(path.dirname(resolvedPath), extends)` is used.

3. **A visited-configs `Set` is threaded through recursive calls** as a safety net: if a config path is visited twice the recursion stops immediately with a log message, so any edge case that still produces a cycle cannot loop forever.

## Tests

A new unit test covers the exact pattern: a tsconfig in a subdirectory uses `"extends": "./tsconfig.json"` to extend its sibling, while the root config extends that subdirectory config. Without the fix the test loops indefinitely (vitest timeout); with the fix it terminates and correctly resolves the path aliases from the sibling config.

All **140 unit tests pass**. The failing integration tests (`cli.spec.ts` watch-mode, `runner.spec.ts`) are **pre-existing** — they require a compiled `dist/bin/cli.js` that is absent from a fresh clone and are unrelated to this change.

### CODE REVIEW

File: packages/skott/src/modules/walkers/ecmascript/typescript/path-alias.ts [hunk 1]
```diff
@@ -154,11 +178,23 @@ export async function buildPathAliases(
...
+      // the tsconfig that declares them, not relative to CWD. Without this,
+      // a package in node_modules that uses a relative extends (e.g.
+      // `@vue/tsconfig/tsconfig.dom.json` extends `./tsconfig.json`) would be
+      // resolved against CWD, looping back to the project root config.
+      const resolvedExtendedPath = extendedTsConfigPath.startsWith(".")
+        ? path.isAbsolute(resolvedPath)
+          ? path.resolve(path.dirname(resolvedPath), extendedTsConfigPath)
+          : path.join(path.dirname(resolvedPath), extendedTsConfigPath)
+        : extendedTsConfigPath;
+
```
Comment (antoine-coulon) [inline packages/skott/src/modules/walkers/ecmascript/typescript/path-alias.ts:191]:
```ts
const resolvedExtendedPath = path.join(
    path.dirname(resolvedPath),
    extendedTsConfigPath
);
```

If I just keep that chunk above (the code you wrote minus ternary branches) the new test you introduced still succeeds. Ideally, each newly introduced branch in that nested ternary has to be justified by a dedicated failing test to act as a documentation on the behavior. Removing any branch of that part should turn at least one test in the red.

Also, note that if I also revert the whole chunk of code and run the test, there is no infinite loop, just a failing test. If you want to go down to the path of a real infinite loop you might want to leverage integration tests, as emulating CWD across in memory file system instances is sometimes harder that just reproducing it with the real file system. To be honest, if you're close enough in the replicated behavior with unit tests, this would make me happy. But I would rather have reliable tests that really reproduce the issue rather than unit tests missing the target. Let me know what you find out

**Review by antoine-coulon — CHANGES_REQUESTED** [review]

**Comment by antoine-coulon** [comment]
Thank you very much @brugmanjoost, solid work!
If we can get to a point where we have tests fully covering the code and the newly introduced behavior (fixing many issues), it would be perfect and would directly be shipped into a new skott version.
